.github/workflows/lintcheck.yml

```diff
@@ -0,0 +1,111 @@
+name: Lintcheck
+
+on: pull_request
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_INCREMENTAL: 0
+
+concurrency:
+  # For a given workflow, if we push to the same PR, cancel all previous builds on that PR.
+  group: "${{ github.workflow }}-${{ github.event.pull_request.number}}"
+  cancel-in-progress: true
+
+jobs:
+  base:
+    runs-on: ubuntu-latest
+
+    outputs:
+      key: ${{ steps.key.outputs.key }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
```

Comment:
I think it would be better to either checkout the head directly, which should be:

```
    steps:
      - uses: actions/checkout@v4
        with:
          ref: ${{ github.base_ref }}
```

Or fetch `master` which should allow us to use `fetch-depth: 0` again

Reply:
`fetch-depth: 0` means the entire history, I ~~stole~~ came up with 2 by reading the checkout action docs, it had an example that checks out `HEAD^` - https://github.com/actions/checkout?tab=readme-ov-file#Checkout-HEAD

There's a potential race condition with using a ref name. If the action gets queued to run on `refs/pull/N/merge` is followed by a push to `master` then `HEAD^` and `master` now refer to separate commits

slightly related: I tested in https://github.com/rust-lang/rust-clippy/pull/12933 and the comment I have about merge conflicts with merges currently is incorrect. If there's a merge conflict you just don't get a `pull_request` workflow run at all until you resolve it, so that's one less thing to worry about

Reply:
Okay, I'm fine with keeping it like this :+1: